sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/QueryBuilderImpl.java

```diff
@@ -0,0 +1,1369 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.common.cql.builder;
+
+import static java.lang.String.format;
+
+import com.github.misberner.apcommons.util.AFModifier;
+import com.github.misberner.duzzt.annotations.DSLAction;
+import com.github.misberner.duzzt.annotations.GenerateEmbeddedDSL;
+import com.github.misberner.duzzt.annotations.SubExpr;
+import io.stargate.sgv2.common.cql.ColumnUtils;
+import io.stargate.sgv2.common.cql.Strings;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/** Convenience builder for creating queries. */
+@GenerateEmbeddedDSL(
+    modifier = AFModifier.DEFAULT,
+    autoVarArgs = false,
+    name = "QueryBuilder",
+    syntax = "(<keyspace>|<table>|<insert>|<update>|<delete>|<select>|<index>|<type>) build",
+    where = {
```

Comment:
Is there any idea how we could rewrite this to something that provides better typed stuff in the middle of the query building? Do we at least have an idea what can this be?